### PR TITLE
Bookstore: initialize project with components

### PR DESCRIPTION
**Pull request by BahirHakimy**
# In this branch I have added the following:
- Setup the project with Create React App
- Created all the components
- Added routing using React Router Dom
- Added Navbar

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,22 @@
-import logo from './logo.svg';
-import './App.css';
+import { createBrowserRouter, RouterProvider } from 'react-router-dom';
+import Layout from 'components/Layout';
+import 'styles/App.css';
```
Comment (mikethreels) [inline src/App.js]: - [ ] due to there not being an App.css in the styles folder the user receives below error when trying to open your application.
afbeelding

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
+  const routes = createBrowserRouter([
+    {
+      path: '/',
+      element: <Layout />,
+      children: [
+        { path: 'books', element: <BookList /> },
+        { path: 'categories', element: <Categories /> },
+      ],
+    },
+  ]);
```
Comment (mikethreels) [inline src/App.js]: - [ ] when opening your application on initial load the user gets an empty screen as you can see in below example.
afbeelding
this is due to the fact that there is not root path defined in your routes, you can solve this issue by making the below adjustment to your code:
```
  const routes = createBrowserRouter([
    {
      path: '/',
      element: <Layout />,
      children: [
        { path: '/', element: <BookList /> },
        { path: 'books', element: <BookList /> },
        { path: 'categories', element: <Categories /> },
      ],
    },
  ]);
```

File: src/components/BookList.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          type="text"
+          name="bookTitle"
+          id="bookTitle"
+          placeholder="Book title"
+          className="bg-white border text-slate-400 px-4 py-2 w-[40rem] border-[#e8e8e8] outline-none text-sm"
+        />
+        <select
+          name="category"
+          id="category"
+          className="bg-white border text-slate-400 px-4 py-2 w-60 border-[#e8e8e8] text-sm "
```
Comment (mikethreels) [inline src/components/BookList.jsx]: - [ ] According to the requirements the part where you add a new book should be in a separate component, you can use the empty AddBook.jsx file you created for this.
- [ ] Also according to the requirements "in the design you can see an input for a category - please replace it with an author."

File: src/components/Categories.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+function Categories() {
+  return (
+    <div>
+      <h1>Categories Page</h1>
+    </div>
+  );
```
Comment (mikethreels) [inline src/components/Categories.jsx]: - [ ] According to the requirements the Categories component should contain a button that checks for the status.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Bahir, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Great job implementing the React router!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.jsx (hunk 4 above)
Comment (BahirHakimy) [reply src/components/Categories.jsx]: Hey @mikethreels, Thanks for your review I will implement all suggested changes, but I didn't understand this part, what kind of button should it check for which type of status? 
Please make it more clear.

File: src/components/BookList.jsx (hunk 3 above)
Comment (BahirHakimy) [reply src/components/BookList.jsx]: @mikethreels There is no point in the requirement that says the add book should be a component, and also in the requirement, there is nothing about changing the category to the author in add book form

File: src/App.js (hunk 1 above)
Comment (BahirHakimy) [reply src/App.js]: @mikethreels Thanks for pointing out, I will make it right

File: src/App.js (hunk 2 above)
Comment (BahirHakimy) [reply src/App.js]: @mikethreels Ok I will do it, Thanks

**Comment by DammyShittu** [comment]
Hi @BahirHakimy,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Your linter checks were running before the previous review but now they are not running. Kindly inspect that and ensure they are running correctly

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/AddBook.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          id="category"
+          className="bg-white border text-slate-400 px-4 py-2 w-60 border-[#e8e8e8] text-sm "
+        >
+          <option value="" selected disabled>
+            Category
+          </option>
+          <option value="action">Action</option>
+          <option value="science fiction">Science Fiction</option>
+          <option value="economy">Economy</option>
+        </select>
```
Comment (danmainah) [inline src/components/AddBook.jsx]: - [ ] @BahirHakimy as previously requested kindly change the category input to the author input, as I have highlighted below it's a project requirement.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @BahirHakimy , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @BahirHakimy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Great implementation based on previous reviewer's request 🥇 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**